Project Walking/Source code/app/src/main/java/com/teamapricot/projectwalking/MainActivity.java

```diff
@@ -1,82 +1,84 @@
+
+    boolean mapInitialized = false;
+
+    private MapView map = null;
 
-    MapView map = null;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        //handle permissions first, before map is created. not depicted here
-
         PermissionHandler permissionHandler = new PermissionHandler(this);
         CameraHandler cameraHandler = new CameraHandler(this);
 
         photoController = new PhotoController(this, permissionHandler, cameraHandler);
 
-
         locationHandler = new LocationHandler(this, 2000);
 
-        locationHandler.registerUpdateListener(position -> {
-            ((TextView)findViewById(R.id.helloWorld))
-                    .setText("(" + position.getLatitude() + "," + position.getLongitude() + ")");
-        });
-
-        //load/initialize the osmdroid configuration, this can be done
         Context ctx = getApplicationContext();
         Configuration.getInstance().load(ctx, PreferenceManager.getDefaultSharedPreferences(ctx));
-        //setting this before the layout is inflated is a good idea
-        //it 'should' ensure that the map has a writable location for the map cache, even without permissions
-        //if no tiles are displayed, you can try overriding the cache path using Configuration.getInstance().setCachePath
-        //see also StorageUtils
-        //note, the load method also sets the HTTP User Agent to your application's package name, abusing osm's tile servers will get you banned based on this string
 
-        //inflate and create the map
         setContentView(R.layout.activity_main);
 
-        map = (MapView) findViewById(R.id.map);
+        map = findViewById(R.id.map);
         map.setTileSource(TileSourceFactory.MAPNIK);
+
+        mapController = map.getController();
+        mapController.setZoom(19.5);
+
+        locationOverlay = new MyLocationNewOverlay(new GpsMyLocationProvider(ctx), map);
+        locationOverlay.enableMyLocation();
+
+        map.getOverlays().add(locationOverlay);
+
+        locationHandler.registerUpdateListener(position -> {
+            GeoPoint point = new GeoPoint(position.getLatitude(), position.getLongitude());
+            if(!mapInitialized) {
+                mapController.setCenter(point);
+                mapInitialized = true;
+            }
+        });
     }
+
     @Override
-    protected void onResume(){
+    protected void onResume() {
         super.onResume();
-        //this will refresh the osmdroid configuration on resuming.
-        //if you make changes to the configuration, use
-        //SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
-        //Configuration.getInstance().load(this, PreferenceManager.getDefaultSharedPreferences(this));
-        map.onResume(); //needed for compass, my location overlays, v6.0.0 and up
+
+        map.onResume();
     }
-    @Override
-    protected void onPause(){
+
+    public void onPause() {
```

Comment:
This should be set to protected, and it should override the method in the base class `Activity`.